### PR TITLE
Multiple servers: finish callbacks when error

### LINKED ISSUE #1659: Cannot rename symbol when eslint is configured
I've enabled eslint alongside typescript with by by setting `lsp-eslint-server-command` in my startup. When I do this and try to rename a symbol via `lsp-rename`, I get a timeout (maybe because it's trying to send the rename to both backends?).

Glad to try and fix this if it is indeed a bug. Pointers would be welcome.

### PR DESCRIPTION
**Pull request by jkdufair**
This is my naive attempt to gracefully complete callbacks when on of the servers returns an error.

Closes #1659

### CODE REVIEW

File: lsp-mode.el [hunk 1]
```diff
@@ -3001,6 +3002,7 @@ If NO-MERGE is non-nil, don't merge the results but return alist workspace->resu
                               nil
                               target-workspaces))
              (error-callback (lambda (error)
+                               (funcall callback nil)
```
Comment (yyoncho) [inline lsp-mode.el:3005]: nil is valid return value so you cannot use it to indicate an error: use something like `:error`.

File: lsp-mode.el [hunk 2]
```diff
@@ -2904,10 +2904,11 @@ MODE determines when the callback will be called depending on the
...
 If NO-MERGE is non-nil, don't merge the results but return alist workspace->result.
 ID is the request id. "
-  (let (results)
+  (let (results errors)
     (lambda (result)
-      (push (cons lsp--cur-workspace result) results)
-      (when (eq (length results) (length workspaces))
+      (push (cons lsp--cur-workspace result)
+            (if result results errors))
+      (when (eq (+ (length errors) (length results)) (length workspaces))
```
Comment (yyoncho) [inline lsp-mode.el:2911]: here we should check if the server has returned only errors and in that case, we should not call the success callback

**Comment by jkdufair** [comment]
@yyoncho I have applied your suggestions. Thank you for those.

**Comment by yyoncho** [comment]
Thank you for contributing to `lsp-mode` and congrats for your first `lsp-mode` PR! :tada:

**Pull request merged**